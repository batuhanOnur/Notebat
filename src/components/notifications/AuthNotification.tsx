import React, { useEffect } from 'react'
import { useNavigate  } from 'react-router-dom'
import { showNotification } from '@mantine/notifications';

type Props = {
  title: string,
  message: string,
  color: string,
  autoClose: number,
  navigateTo: string
}


const AuthNotification: React.FC<Props> = (props) => {

  const navigate = useNavigate();

  useEffect(() => {
    showNotification({
      title: props.title,
      message: props.message,
      color: props.color,
      autoClose: props.autoClose,
    });

    setTimeout(() => {
       navigate(`/${props.navigateTo}`)
    }, props.autoClose)
    // eslint-disable-next-line react-hooks/exhaustive-deps
  },[])


  return (
    <>
    </>
  )
}

export default AuthNotification