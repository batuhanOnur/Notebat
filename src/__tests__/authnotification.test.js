import renderer from 'react-test-renderer';
import AuthNotification from '../components/notifications/AuthNotification'

// mock useNavigate
const mockedUsedNavigate = jest.fn()

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
   useNavigate: () => mockedUsedNavigate,
 }));

it('Success create notification', () => {
    renderer.create(
     <AuthNotification 
     title={"Login Succesful"}
     message={"Redirect to Home Page"}
     color={"green"}
     autoClose={2000}
     navigateTo={"boards"}
     />
  );
  
});