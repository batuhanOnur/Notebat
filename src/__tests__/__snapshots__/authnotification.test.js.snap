// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Success create notification 1`] = `null`;
